include/AST/Types/ArrayTy.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef GLU_AST_TYPES_ARRAYTY_HPP_
+#define GLU_AST_TYPES_ARRAYTY_HPP_
+
+#include "Types/TypeBase.hpp"
+#include <optional>
+
+namespace glu::types {
+
+/// @brief Represents an array type.
+class ArrayTy : public TypeBase {
+    TypeKind const _dataKind;
+    enum ArrayMode { Dynamic, Static } _mode;
```

Comment:
I think that static arrays and dynamic arrays should be different kinds of types